### PR TITLE
IDE-4324 Buildship upgrade to 3.x latest

### PR DESCRIPTION
**Pull request by simonjhy**

### CODE REVIEW

File: tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleCore.java [hunk 1]
```diff
@@ -38,10 +38,12 @@
  * @author Andy Wu
  */
 @SuppressWarnings("restriction")
-public class GradleCore extends Plugin {
+public class LiferayGradleCore extends Plugin {
 
 	public static final String LIFERAY_WATCH = "liferay-watch";
 
+	public static final String LIFERAY_WATCH_DEPLOY_TASK = "Liferay Watch Task";
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleCore.java]: lets remove this

Comment (simonjhy) [reply tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleCore.java]: I removed this and resent pull.

File: tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java [hunk 2]
```diff
@@ -166,16 +165,15 @@ public IStatus createNewProject(NewLiferayModuleProjectOp op, IProgressMonitor m
 			}
 
 			if ((hasGradleWorkspace && useDefaultLocation) || inWorkspacePath) {
+				GradleUtil.sychronizeProject(projectLocation, monitor);
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java]: we add GradleUtil.sychronizeProject in both if and else blocks, does it mean we should run it always? if yes, we should put it out of if block.

File: tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleUtil.java [hunk 3]
```diff
@@ -309,7 +267,7 @@ private static GradleRunConfigurationAttributes _getRunConfigurationAttributes(
 
 		GradleDistribution gradleDistribution = buildConfig.getGradleDistribution();
 
-		String serializeString = gradleDistribution.serializeToString();
+		String serializeString = gradleDistribution.toString();
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleUtil.java]: why do we need to change this?

**Comment by jtydhr88** [comment]
see comments and resend, thanks.

**Pull request closed without merging**